src/test/java/uk/gov/hmcts/cmc/claimstore/services/CountyCourtJudgmentServiceTest.java

```diff
@@ -107,7 +109,31 @@ public void saveShouldFinishCCJByAdmissionSuccessfullyForHappyPath() {
 
         verify(eventProducer, once()).createCountyCourtJudgmentEvent(any(Claim.class), any());
         verify(claimService, once()).saveCountyCourtJudgment(eq(AUTHORISATION), any(), any());
-        verify(appInsights, once()).trackEvent(eq(AppInsightsEvent.CCJ_REQUESTED),
+        verify(appInsights, once()).trackEvent(eq(AppInsightsEvent.CCJ_REQUESTED_BY_ADMISSION),
+            eq(REFERENCE_NUMBER), eq(claim.getReferenceNumber()));
+    }
+
+    @Test
+    public void shouldSaveCCJAfterBreachOfSettlementAgreement() {
+
+        Settlement settlement = new Settlement();
+        settlement.makeOffer(SampleOffer.builderWithSetByDateInPast().build(), MadeBy.DEFENDANT);
+        settlement.accept(MadeBy.CLAIMANT);
+
+        Claim claim = SampleClaim
+            .builder()
+            .withResponse(SampleResponse.FullAdmission.builder().build())
+            .withSettlement(settlement)
+            .build();
+
+        when(claimService.getClaimByExternalId(eq(EXTERNAL_ID), eq(AUTHORISATION))).thenReturn(claim);
+
+        CountyCourtJudgment ccjByAdmission = SampleCountyCourtJudgment.builder().ccjType(ADMISSIONS).build();
```

Comment:
Is it worth putting InOrder like we do in other tests so that order invocations can be asserted?